connectors/sources/servicenow.py

```diff
@@ -470,15 +530,101 @@ def get_default_configuration(cls):
                 "ui_restrictions": ["advanced"],
                 "value": False,
             },
+            "use_document_level_security": {
+                "display": "toggle",
+                "label": "Enable document level security",
+                "order": 8,
+                "tooltip": "Document level security ensures identities and permissions set in ServiceNow are maintained in Elasticsearch. This enables you to restrict and personalize read-access users and groups have to documents in this index. Access control syncs ensure this metadata is kept up to date in your Elasticsearch documents.",
+                "type": "bool",
+                "value": False,
+            },
+            "service_roles_mapping": {
+                "depends_on": [{"field": "use_document_level_security", "value": True}],
+                "display": "textarea",
+                "label": "Mapping of Roles and Services",
+                "order": 9,
+                "type": "str",
+            },
         }
 
+    def _dls_enabled(self):
+        """Check if document level security is enabled. This method checks whether document level security (DLS) is enabled based on the provided configuration.
+
+        Returns:
+            bool: True if document level security is enabled, False otherwise.
+        """
+        if (
+            self._features is None
+            or not self._features.document_level_security_enabled()
+        ):
+            return False
+
+        return self.configuration["use_document_level_security"]
+
+    async def _user_access_control_doc(self, user):
+        user_id = user.get("_id", "")
```

Comment:
I'm not sure, if this will work. If the `user_id` is empty the framework will probably use an empty `_id` when ingesting the doc, which would lead to Elasticsearch auto-generating a value for `_id`. So for every access control sync N new documents would be added (which conceptually refer to the same identity), if you've N documents without an `_id`.

Can you verify that behavior? If that's the case we should probably log a warning, if the `_id` is missing without yielding it to the framework.